lightly_studio/src/lightly_studio/dataset/embedding_manager.py

```diff
@@ -94,48 +96,55 @@ def register_embedding_model(
         self._models[model_id] = embedding_generator
 
         # Set as default if requested or if it's the first model
-        if set_as_default or self._default_model_id is None:
-            self._default_model_id = model_id
+        if set_as_default or self._default_model_id.get(dataset_id, None) is None:
+            self._default_model_id[dataset_id] = model_id
 
         return db_model
 
-    def embed_text(self, text_query: TextEmbedQuery) -> list[float]:
+    def embed_text(
+        self,
+        dataset_id: UUID,
+        text_query: TextEmbedQuery,
+    ) -> list[float]:
         """Generate an embedding for a text sample.
 
         Args:
+            dataset_id: The ID of the dataset to associate with the model.
```

Comment:
The comment should be expanded (multiple sentences are ok). I though this is true but it is not accurate either.
```suggestion
            dataset_id: The ID of the dataset associated with the model.
```